server/enhancedDate.js

```diff
@@ -1 +1,103 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+// export this as a Node module using the enhancedDate.js code from Module_2
+
+
+'use strict';
+var dateObjects = {
+  dateTime: '',
+  monthNames: ['January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December'],
+  dayNames: ['Sunday',
+    'Monday',
+    'Tuesday',
+    'Wednesday',
+    'Thursday',
+    'Friday',
+    'Saturday'],
+  milsDay: 86400000
+};
+
+var now = function () {
+  return new Date();
+};
+
+var setDate = function (e) {
+    if (e instanceof Date) {
+      dateObjects.dateTime = e;

```

Comment:
Better to pass e in to the Date constructor to prevent external changes messing with your internal data.